google/auth/transport/aiohttp_req.py

```diff
@@ -0,0 +1,337 @@
+            to make HTTP requests. If not specified, a session will be created.
+
+    .. automethod:: __call__
+    """
+
+    def __init__(self, session=None):
+        if not session:
+            session = aiohttp.ClientSession()
+        self.session = session
+
+    async def __call__(
+        self,
+        url,
+        method="GET",
+        body=None,
+        headers=None,
+        timeout=_DEFAULT_TIMEOUT,
+        **kwargs
+    ):
+        """Make an HTTP request using aiohttp.
+
+        Args:
+            url (str): The URI to be requested.
+            method (str): The HTTP method to use for the request. Defaults
+                to 'GET'.
+            body (bytes): The payload / body in HTTP request.
+            headers (Mapping[str, str]): Request headers.
+            timeout (Optional[int]): The number of seconds to wait for a
+                response from the server. If not specified or if None, the
+                requests default timeout will be used.
+            kwargs: Additional arguments passed through to the underlying
+                requests :meth:`~requests.Session.request` method.
+
+        Returns:
+            google.auth.transport.Response: The HTTP response.
+
+        Raises:
+            google.auth.exceptions.TransportError: If any exception occurred.
+        """
+        try:
+            _LOGGER.debug("Making request: %s %s", method, url)
+            response = await self.session.request(
+                method, url, data=body, headers=headers, timeout=timeout, **kwargs
+            )
+            return _Response(response)
+
+        except aiohttp.ClientError as caught_exc:
+            new_exc = exceptions.TransportError(caught_exc)
+            six.raise_from(new_exc, caught_exc)
+
+        except asyncio.TimeoutError as caught_exc1:
+            new_exc1 = exceptions.TransportError(caught_exc1)
+            six.raise_from(new_exc1, caught_exc1)
+
+
+class AuthorizedSession(aiohttp.ClientSession):
+    """
+
+    documentation
+
+    """
+
+    def __init__(
+        self,
+        credentials,
+        refresh_status_codes=transport.DEFAULT_REFRESH_STATUS_CODES,
+        max_refresh_attempts=transport.DEFAULT_MAX_REFRESH_ATTEMPTS,
+        refresh_timeout=None,
+        auth_request=None,
+    ):
+        super(AuthorizedSession, self).__init__()
+        self.credentials = credentials
+        self._refresh_status_codes = refresh_status_codes
+        self._max_refresh_attempts = max_refresh_attempts
+        self._refresh_timeout = refresh_timeout
+        self._is_mtls = False
+        self._auth_request = auth_request
+        self._auth_request_session = None
+        self._loop = asyncio.get_event_loop()
+        self._refresh_lock = asyncio.Lock()
+
+        if auth_request is None:
+            self._auth_request_session = aiohttp.ClientSession()
+            auth_request = Request(self._auth_request_session)
+            self._auth_request = auth_request
```

Comment:
I notice there is no use of a retry adapter here, which there is in the non-async version. Will there be retries on network errors here?  If not, is there a way to include this functionality?

Reply:
The adapters were only compatible with sync requests and I wasn't able to find an async aiohttp equivalent, so ended up removing the retry logic - I think that if we are able to find an alternative to add this extra check to network errors that would definitely be good. 

My reasoning for removing it is that even with the adapter logic there isn't a guarantee of always creating a network connection (the probability of not doing so with perfect independence of each retry is just cubed ^3 and the choice of 3 seemed arbitrary), so without the guarantee I felt that functionally we weren't fundamentally doing something different with the retry adapter. Adding it would be good though.